likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/universalfeed/view/LMFeedUniversalFeedListView.kt

```diff
@@ -0,0 +1,154 @@
+package com.likeminds.feed.android.core.universalfeed.view
+
+import android.content.Context
+import android.util.AttributeSet
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.*
+import com.likeminds.feed.android.core.R
+import com.likeminds.feed.android.core.universalfeed.adapter.LMFeedUniversalFeedAdapter
+import com.likeminds.feed.android.core.universalfeed.adapter.LMFeedUniversalFeedAdapterListener
+import com.likeminds.feed.android.core.universalfeed.model.LMFeedPostViewData
+import com.likeminds.feed.android.core.utils.*
+
+class LMFeedUniversalFeedListView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : RecyclerView(context, attrs, defStyleAttr) {
+
+    val linearLayoutManager: LinearLayoutManager
+    private val dividerDecoration = DividerItemDecoration(context, DividerItemDecoration.VERTICAL)
+
+    private lateinit var universalFeedAdapter: LMFeedUniversalFeedAdapter
+    private lateinit var postVideoAutoPlayHelper: LMFeedPostVideoAutoPlayHelper
+    private lateinit var paginationScrollListener: LMFeedEndlessRecyclerViewScrollListener
+
+    val itemCount: Int get() = universalFeedAdapter.itemCount
+
+    init {
+        setHasFixedSize(true)
+        linearLayoutManager = LinearLayoutManager(context)
+        layoutManager = linearLayoutManager
+
+        if (itemAnimator is SimpleItemAnimator)
+            (itemAnimator as SimpleItemAnimator).supportsChangeAnimations = false
+
+        //item decorator to add spacing between items
+        val dividerDrawable = ContextCompat.getDrawable(
+            context,
+            R.drawable.lm_feed_item_divider
+        )
+        dividerDrawable?.let { drawable ->
+            dividerDecoration.setDrawable(drawable)
+        }
+
+        addItemDecoration(dividerDecoration)
+
+        //todo: testing required
+        initiateAutoPlayer()
```

Comment:
todo , have we tested?

Reply:
yet to be tested, will remove this todo once testing is done